spec/API_specification/searching_functions.md

```diff
@@ -69,6 +69,12 @@ Returns the indices of the minimum values along a specified axis. When the minim
 (function-nonzero)=
 ### nonzero(x, /)
 
+:::{admonition} Data-dependent output shape
+:class: important
+
+The shape of the output array for this function depends on the data values in the input array, hence it can be difficult to implement for libraries that build computation graphs for arrays without knowing their values. See {ref}`indexing` section for more details.
```

Comment:
Not clear to me what we are referring to in the "indexing section". I think, instead, what is want is to refer to a separate page under "Design topics" where we specifically discuss APIs which return arrays having data-dependent shapes (for further discussion, see [here](https://github.com/data-apis/array-api/pull/168#pullrequestreview-650684682)).

Reply:
I think we need to add a new section in "Design Topics" (via this PR) and then reference that section.